### PR TITLE
Add tests for disabled syncplan logic

### PR DESCRIPTION
**Pull request by swadeley**
Hello

adding a test to ensure a cancelled recurring logic does not prevent an upgrade or disable the previously associated sync plan.

 Closed loop: BZ#1887511

### CODE REVIEW

File: tests/upgrades/test_syncplan.py [hunk 1]
```diff
@@ -56,6 +56,49 @@ def test_pre_sync_plan_migration(self, request):
...
+            organization=org, name=f'{request.node.name}_syncplan', interval="weekly"
+        ).create()
+        product = entities.Product(organization=org, name=f'{request.node.name}_prod').create()
+        entities.Repository(product=product, name=f'{request.node.name}_repos').create()
+        sync_plan.add_products(data={'product_ids': [product.id]})
+        product.sync()
+        product = product.read()
+        assert product.sync_plan.id == sync_plan.id
+        # Note the recurring logic ID for later assert a new one was created
+        count = sync_plan.foreman_tasks_recurring_logic.id
```
Comment (pondrejk) [inline tests/upgrades/test_syncplan.py:90]: not sure why we assign `count = 0` on line 78, did you plan to do some assert here?

File: tests/upgrades/test_syncplan.py [hunk 2]
```diff
@@ -56,6 +56,49 @@ def test_pre_sync_plan_migration(self, request):
...
+        # Cancel the recurring logic
+        entities.RecurringLogic(id=count).read()
+        entities.RecurringLogic(id=count).cancel()
+        # Re-enable the sync plan (it will get a new recurring logic)
+        sync_plan.enabled = True
+        sync_plan.update(['enabled'])
+        sync_plan = sync_plan.read()
+        assert sync_plan.enabled
+        # Assert a new recurring logic was assigned
+        assert sync_plan.foreman_tasks_recurring_logic.id > count
```
Comment (pondrejk) [inline tests/upgrades/test_syncplan.py:100]: why rely on the fact that IDs represent count, doing just new_id != old_id would be more change-proof

**Review by pondrejk — APPROVED** [review]

**Comment by pondrejk** [comment]
minor comments

File: tests/upgrades/test_syncplan.py (hunk 2 above)
Comment (mshriver) [reply tests/upgrades/test_syncplan.py:100]: I'd agree here, calling it `count` is misleading.

Comment (swadeley) [reply tests/upgrades/test_syncplan.py:100]: I guess I chose "count" as the sync plan ID is incremented, will change it as suggested

Comment (swadeley) [reply tests/upgrades/test_syncplan.py:100]: As this was already merged, I created new PR https://github.com/SatelliteQE/robottelo/pull/9125

**Pull request merged**